src/arrayMethodSort.js

```diff
@@ -4,8 +4,34 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = (a, b) => (a.toString() > b.toString() ? 1 : -1),
+  ) {
+    let thisLenght = this.length;
+
+    while (thisLenght > 0) {
+      for (let i = 0; i < thisLenght - 1; i++) {
+        const current = this[i];
+        const next = this[i + 1];
+
+        if (compareFunction(this[i], this[i + 1])) {
+          if (compareFunction(this[i], this[i + 1]) > 0) {
+            this[i] = next;
+            this[i + 1] = current;
+          } else {
+            continue;
+          }
+        } else {
+          if (this[i] > this[i + 1]) {
+            this[i] = next;
+            this[i + 1] = current;
+          }
+        }
```

Comment:
The condition `compareFunction(this[i], this[i + 1])` is used to check if a swap is needed, but it's being called twice which is unnecessary. You should call it once and store the result in a variable to avoid calling the same function multiple times with the same arguments. Also, the `else` block starting at line 24 is unnecessary because the `continue` statement on line 22 prevents any further execution if the condition on line 17 is not met.